mm2src/coins/eth.rs

```diff
@@ -3017,18 +3015,18 @@ impl RpcCommonOps for EthCoin {
 
         // try to find first live client
         for (i, client) in clients.clone().into_iter().enumerate() {
-            if let Web3Transport::Websocket(socket_transport) = &client.web3.transport() {
+            if let Web3Transport::Websocket(socket_transport) = client.as_ref().transport() {
                 socket_transport.maybe_spawn_connection_loop(self.clone());
             };
 
-            if !client.web3.transport().is_last_request_failed() {
+            if client.as_ref().transport().is_last_request_failed() {
```

Comment:
why remove the not condition, was it a bug?

Reply:
no, it's my mistake